notebooks/systemds_dev.ipynb

```diff
@@ -0,0 +1,882 @@
+            "Maven home: /opt/apache-maven-3.6.3\n",
+            "Java version: 1.8.0_252, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre\n",
+            "Default locale: en_US, platform encoding: UTF-8\n",
+            "OS name: \"linux\", version: \"4.19.104+\", arch: \"amd64\", family: \"unix\"\n",
+            "\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Xphbe3R43XLw",
+        "colab_type": "text"
+      },
+      "source": [
+        "#### Download Apache Spark"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_WgEa00pTs3w",
+        "colab_type": "text"
+      },
+      "source": [
+        "NOTE: If spark is not downloaded. Let us make sure the version we are trying to download is officially supported at\n",
+        "https://spark.apache.org/downloads.html"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "3zdtkFkLnskx",
+        "colab_type": "code",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 119
+        },
+        "outputId": "a5b8b7ef-a401-4549-e521-320f9f22ed41"
+      },
+      "source": [
+        "# Spark and Hadoop version\n",
+        "spark_version = 'spark-2.4.6'\n",
+        "hadoop_version = 'hadoop2.7'\n",
+        "spark_path = f\"/opt/{spark_version}-bin-{hadoop_version}\"\n",
+        "if not os.path.exists(spark_path):\n",
+        "  run(f\"wget -q -nc -O apache-spark.tgz https://downloads.apache.org/spark/{spark_version}/{spark_version}-bin-{hadoop_version}.tgz\")\n",
+        "  run('tar zxf apache-spark.tgz -C /opt')\n",
+        "  run('rm -f apache-spark.tgz')\n",
+        "\n",
+        "os.environ[\"SPARK_HOME\"] = spark_path\n",
+        "os.environ[\"PATH\"] += \":$SPARK_HOME/bin\"\n"
+      ],
+      "execution_count": 13,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            ">> wget -q -nc -O apache-spark.tgz https://downloads.apache.org/spark/spark-2.4.6/spark-2.4.6-bin-hadoop2.7.tgz\n",
+            "\n",
+            ">> tar zxf apache-spark.tgz -C /opt\n",
+            "\n",
+            ">> rm -f apache-spark.tgz\n",
+            "\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "91pJ5U8k3cjk",
+        "colab_type": "text"
+      },
+      "source": [
+        "#### Get Apache SystemDS"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "SaPIprmg3lKE",
+        "colab_type": "code",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 153
+        },
+        "outputId": "1eee45c1-cd92-480e-d27c-a2229086b5a0"
+      },
+      "source": [
+        "!git clone https://github.com/apache/systemds systemds\n",
```

Comment:
Could we use the trick of only downloading the last commit `--depth=1`

Reply:
Yes, will do.